pkg/api/validation/validation.go

```diff
@@ -157,8 +157,18 @@ func validateOwnerReference(ownerReference api.OwnerReference, fldPath *field.Pa
 
 func ValidateOwnerReferences(ownerReferences []api.OwnerReference, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+	var controllerRef *api.OwnerReference
 	for _, ref := range ownerReferences {
 		allErrs = append(allErrs, validateOwnerReference(ref, fldPath)...)
+		if ref.Controller != nil && *ref.Controller {
+			if controllerRef != nil {
+				allErrs = append(allErrs, field.Invalid(fldPath, ownerReferences,
+					fmt.Sprintf("Only one reference can have Controller set to true. Found \"true\" in references for %v and %v", controllerRef.Name, ref.Name)))
+			} else {
+				refCopy := ref
+				controllerRef = &refCopy
```

Comment:
Why do you need to record the entire controllerRef here? Can you just use a bool and a string to record the controllerRef.Name?